From 80e8ddc0fa043d005209e481fb82674fd271d72c Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Thu, 24 Apr 2025 10:55:09 +0200
Subject: [RISE 04/10] crypto: stm32: fix stm32_saes CTR mode on small input
 buffers

Fix missing cast when saving pre-computed masks in STM32 SAES driver
CTR implementation when several small input data is provided to
the update handler.

The issue is revealed by xtest regression_4017 when run with at least
level 12, e.g. "xtest -l 15 regression_4017".

Fixes: 4320f5cf30c5 ("crypto: stm32: SAES cipher support")
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Acked-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Acked-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Acked-by: Jens Wiklander <jens.wiklander@linaro.org>
---
 core/drivers/crypto/stm32/stm32_saes.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/drivers/crypto/stm32/stm32_saes.c b/core/drivers/crypto/stm32/stm32_saes.c
index 47a1fceda23d..298dfecde3e5 100644
--- a/core/drivers/crypto/stm32/stm32_saes.c
+++ b/core/drivers/crypto/stm32/stm32_saes.c
@@ -1035,7 +1035,7 @@ TEE_Result stm32_saes_update(struct stm32_saes_context *ctx, bool last_block,
 			 */
 
 			/* We save remaining mask and its new size */
-			memmove(ctx->extra, ctx->extra + j,
+			memmove(ctx->extra, (uint8_t *)ctx->extra + j,
 				ctx->extra_size - j);
 			ctx->extra_size -= j;
 
-- 
2.43.0

